### PR TITLE
Support poc mode

### PR DESCRIPTION
**Pull request by NoamALK**

### CODE REVIEW

File: src/aws_ec2_auto_onboarding/AWSEc2AutoOnboarding.py [hunk 1]
```diff
@@ -1,10 +1,12 @@
 import urllib3
-import pvwa_integration
+from pvwa_integration import pvwa_integration
 import aws_services
 import instance_processing
 import pvwa_api_calls
 import json
 
+IS_SAFE_HANDLER = False
```
Comment (ryaniv) [inline src/aws_ec2_auto_onboarding/AWSEc2AutoOnboarding.py:8]: not sure it is needed here, if you use it just in the next line

File: src/aws_ec2_auto_onboarding/AWSEc2AutoOnboarding.py [hunk 2]
```diff
@@ -81,14 +83,15 @@ def lambda_handler(event, context):
...
             return
-        # Save PVWA Verification key in /tmp folder
-        crt = open("/tmp/server.crt","w+")
-        crt.write(storeParametersClass.pvwaVerificationKey)
-        crt.close()
+        if storeParametersClass.AOB_mode == 'Production':
+            # Save PVWA Verification key in /tmp folder
+            crt = open("/tmp/server.crt","w+")
+            crt.write(storeParametersClass.pvwaVerificationKey)
+            crt.close()
```
Comment (ryaniv) [inline src/aws_ec2_auto_onboarding/AWSEc2AutoOnboarding.py:90]: better way is to write it as:
with open("/tmp/server.crt","w+") as crt:
    crt.write(storeParametersClass.pvwaVerificationKey)

# it better handling for errors

Comment (ryaniv) [inline src/aws_ec2_auto_onboarding/AWSEc2AutoOnboarding.py:90]: better way is to write it as """"
with open("/tmp/server.crt","w+") as crt:
   crt.write(storeParametersClass.pvwaVerificationKey)
"""

File: src/aws_environment_setup/AWSEnvironmentSetup.py [hunk 3]
```diff
@@ -6,8 +6,12 @@
 import time
 import boto3
 import json
+from pvwa_integration import pvwa_integration
+import aws_services
 from dynamo_lock import LockerClient
 
+IS_SAFE_HANDLER = True
```
Comment (ryaniv) [inline src/aws_environment_setup/AWSEnvironmentSetup.py:13]: dont think you even use it here

File: src/aws_environment_setup/AWSEnvironmentSetup.py [hunk 4]
```diff
@@ -41,18 +45,31 @@ def lambda_handler(event, context):
...
-                return cfnresponse.send(event, context, cfnresponse.FAILED, "Failed to create PVWA Verification Key in Parameter Store",
-                                        {}, physicalResourceId)
-
-            pvwaSessionId = logon_pvwa(requestUsername, requestPassword, requestPvwaIp)
+            
+            if requestS3BucketName == '' and requestVerificationKeyName != '':
+                raise Exception('Verification Key cannot be empty if S3 Bucket is provided')    
+            elif requestS3BucketName != '' and requestVerificationKeyName == '':
+                raise Exception('S3 Bucket cannot be empty if Verification Key is provided')
+            else:
```
Comment (ryaniv) [inline src/aws_environment_setup/AWSEnvironmentSetup.py:60]: this "else" is a bit redundant

File: Jenkinsfile [hunk 5]
```diff
@@ -43,77 +43,81 @@ pipeline {
...
+                     cp -R src/shared_libraries/* src/aws_ec2_auto_onboarding
+                     ls src/aws_ec2_auto_onboarding
+                     cd src/aws_ec2_auto_onboarding
+                     cd package
+                     zip -r9 ${OLDPWD}/aws_ec2_auto_onboarding.zip .
+                     cd $OLDPWD
+                     zip -g aws_ec2_auto_onboarding.zip aws_services.py AWSEc2AutoOnboarding.py instance_processing.py kp_processing.py pvwa_api_calls.py pvwa_integration.py puttygen log_mechanisem.py
+                 '''
+             }
+        }
```
Comment (erz4) [inline Jenkinsfile:76]: you should create function for that

File: src/shared_libraries/instance_processing.py [hunk 6]
```diff
@@ -23,7 +28,7 @@ def delete_instance(instanceId, session, storeParametersClass, instanceData, ins
                                                                             safeName, instanceId,
                                                                             storeParametersClass.pvwaURL)
     if not instanceAccountId:
-        print("Instance AccountId not found on safe")
+        logger.info(instanceId + " was does not exist in safe")
```
Comment (erz4) [inline src/shared_libraries/instance_processing.py]: English very bad

File: src/shared_libraries/instance_processing.py [hunk 7]
```diff
@@ -33,47 +38,52 @@ def delete_instance(instanceId, session, storeParametersClass, instanceData, ins
...
+    	instancePasswordData = ec2Resource.get_password_data(InstanceId=instanceId)
+    	return instancePasswordData['PasswordData']
+    except Exception as e:
+    	logger.error('Error on waiting for instance password: {0}'.format(e))
 
 
 def create_instance(instanceId, instanceDetails, storeParametersClass, logName, solutionAccountId, eventRegion, eventAccountId, instanceAccountPassword):
-
-
+    logger.trace(instanceId, instanceDetails, storeParametersClass, logName, solutionAccountId, eventRegion, eventAccountId, instanceAccountPassword,
```
Comment (erz4) [inline src/shared_libraries/instance_processing.py]: print password? instanceAccountPassword

File: src/shared_libraries/pvwa_api_calls.py [hunk 8]
```diff
@@ -92,10 +108,12 @@ def check_if_kp_exists(session, accountName, safeName, instanceId, restURL):
     else:  # has no value
         pvwaUrl = "{0}/api/accounts?search={1}".format(restURL, accountName)
     try:
-        restResponse = pvwa_integration.call_rest_api_get(pvwaUrl, header)
+        restResponse = pvwa_integration_class.call_rest_api_get(pvwaUrl, header)
         if not restResponse:
+            logger.error("Unknown Error when calling rest service - retrieve accountId")
```
Comment (erz4) [inline src/shared_libraries/pvwa_api_calls.py:113]: duplicate?

**Pull request merged**